File: go/1176.go
package main

import "fmt"

func fastFibonacci(N int) int {
	if N <= 1 {
		return N
	}

	a, b := 0, 1
	for i := 2; i <= N; i++ {
		a, b = b, a+b
	}
	return b
}

func main() {
	var n, x int

	fmt.Scanf("%d", &n)

	for i := 0; i < n; i++ {
		fmt.Scanf("%d", &x)
		fmt.Printf("Fib(%d) = %d\n", x, fastFibonacci(x))
	}
}

File: lua/1176.lua
local function fastFibonacci(N)
    if N <= 1 then
        return N
    end

    local a, b = 0, 1
    for i = 2, N do
        a, b = b, a + b
    end
    return b
end

local n = io.read("*n")

for i = 1, n do
    local x = io.read("*n")
    io.write(string.format("Fib(%d) = %d\n", x, fastFibonacci(x)))
end
